### DISCUSSION TITLE
How to add Geolocation filter (PostGIS e.g.) to default strapi filters while fetching a content data

### DISCUSSION THREAD

**Discussion by Mehrdad-V** [strapi/strapi#6260] [Legacy Topics]
Hello, I want to filter my content by distance in addition to other filters that strapi let us use, but as I searched the only way is to use raw queries which doesn't let me apply other filters.
suppose we have a CITY content with fields NAME, LAT, LONG, POPULATION.
How should I filter by distance and for example population simultaneously?
(IF I don't want to pass other filters in raw query)

**Comment by lauriejim** [comment]
Hello!

Here are defined available filter - [code](https://github.com/strapi/strapi/blob/8281ed7a8fd3f35af9106b846b41d3404c9208d5/packages/strapi-connector-bookshelf/lib/get-query-params.js#L43)
And here is where it's used - [code](https://github.com/strapi/strapi/blob/f2663eb1f7a5d32a23275e182f1c9a52ce2bca3a/packages/strapi-utils/lib/models.js#L474)

These modules, connector and utils are not customizable. Plugins are but not these.
So you will not be able to add one in your project to manage your specific PostGIS filtering.

Strapi is using Bookshelf to manage database connection and to query data.
You will have to create you own logic to manage this kind of filtering.

**Reply by Mehrdad-V** [reply]
thank you so much, I solved my problem with knex.raw in queryBuild in my model's custom controller

**Reply by hadynz** [reply]
@Mehrdad-V any chance you can share a gist with your solution? I need to do something similar and starting off by reading what's out there.
